service/src/test/java/bio/terra/catalog/service/DatasetServiceTest.java

```diff
@@ -187,29 +189,36 @@ void testGetMetadataUsingTdrPermission() {
   void testUpdateMetadataWithInvalidUser() {
     mockDataset();
     assertThrows(
-        UnauthorizedException.class, () -> datasetService.updateMetadata(user, datasetId, "test"));
+        UnauthorizedException.class, () -> datasetService.updateMetadata(user, datasetId, metadata));
   }
 
   @Test
   void testUpdateMetadata() {
     mockDataset();
-    String metadata = "test metadata";
     when(samService.hasGlobalAction(user, SamAction.UPDATE_ANY_METADATA)).thenReturn(true);
     datasetService.updateMetadata(user, datasetId, metadata);
     verify(datasetDao).update(dataset.withMetadata(metadata));
   }
 
+  @Test
+  void testUpdateMetadataInvalidInput() {
+    String metadata = "metadata must be json object";
+    assertThrows(
+        BadRequestException.class, () -> datasetService.updateMetadata(user, datasetId, metadata));
+    verifyNoInteractions(samService);
```

Comment:
I don't think this assert belongs here, since the correctness of the code doesn't depend on if SAM is used or not. You can imagine a version of `updateMetadata()` which checked permission first before checking for valid JSON. That's different than `verifyNoInteractions(datasetDao)`, since we want to make sure that no database changes are made if the json is invalid.

I see that the DAO is used for both `retrieve()` and `update()`, a more correct assert would only affect `update()`. Something like `verify(datasetDao, never()).update();` or `then(datasetDao).should(never()).update();`.